### PR TITLE
[REFACTOR] Fixed Register Survey bug

### PR DESCRIPTION
**Pull request by zikrya**

### CODE REVIEW

**Comment by ravenclaw1999** [comment]
I think there is something off with the profile route. It doesn't seem to be loading. Usually I have to click on it a bunch of times to get the page to show up.

**Comment by aappacode** [comment]
> I think there is something off with the profile route. It doesn't seem to be loading. Usually I have to click on it a bunch of times to get the page to show up.

I'm running into the same issue. It seems though when I add anything/update, it works fine, but its only the initial load it seems like it's checking something for a while

File: src/components/useProtectedRoute.js [hunk 1]
```diff
@@ -7,17 +7,23 @@ export const useProtectedRoute = () => {
   const navigate = useNavigate();
   const location = useLocation();
 
-
   useEffect(() => {
-    if (!loading && !currentUser) {
+    // Avoid navigation if data is still loading
+    if (loading) return;
+
+    // Conditions to avoid unnecessary navigation
```
Comment (aappacode) [inline src/components/useProtectedRoute.js:14]: This fixed the navigation issue with loading!

**Review by aappacode — APPROVED** [review]
Good to merge!

**Pull request merged**